### PR TITLE
[saic_ismart][wip] initial contribution of SAIC iSmart based vehicles

### PR DESCRIPTION
**Pull request by tisoft**
**Work in progress.**

Supports reading out vehicle data of cars running the iSMART system from SAIC.
Currently tested against a MG5 2022.

The API itself is reverse engineered in this project: https://github.com/ReverseEngineeringDE/SAIC-API-Documentation/

If you use this plugin, the iSmart iOS/Android app will be disconnected, since only one login session is supported by the API.

Currently you need to use the official iSmart app to register an account and bind your vehicle.

You should activate "Engine Start Messages" in the iSmart App. The plugin only queries the car while it is running or charging (That is when the HV battery is powered on). This setting will send a message over the API, every time the car is started, so the plugin can detect that without querying the car. This helps to protect from 12V-battery drainage. Additionally there is an advanced "Force refresh" channel, that can be used to manually force an update of the car data. The plugin will query the car 15 minutes after the last activity has been detected. So if you shutdown your engine and start charging within 15 Minutes, the plugin will stay active and display charging data. If you need more than 15 minutes, the charging will not be detected.

- Get vehicle Data
  - [x] Location
  - [x] Speed
  - [x] Mileage
  - [x] SOC
  - [x] Temperatures
  - [x] Door State
  - [x] Window State (if supported by car)
  - [x] Tyre Pressure
  - [x] Charge State
  - [x] current power usage
  - [x] auxiliary battery voltage
  - [ ] Display car messages
  - [ ] accumulated power usage ❓ 
  - [ ] SOH ❓ 
- Control vehicle
  - [ ] Remote Climate Control 🚧 
  - [ ] (Un-)Lock
  - [ ] Locate Car (Light/Honk)
  - [ ] Window Control (if supported by car)
- Generic Features
  - [ ] Relogin after logout/token expiration
  - [ ] Automatically register for all alarm/car messages
  - [ ] Configure available channels based on car capabilities
  - [ ] Display account messages
  - [x] Send data to A better Routeplanner (optional)

❓: It is currently unknown if this feature is possible
🚧: Feature is currently WIP, use with caution

~~**This plugin can drain your 12v battery! it is not yet optimized for power saving!**~~
This should be fixed. I have the plugin running for quite some time and the 12V battery is fine. But please check this regularly.

Download: https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.saicismart/3.4.1-SNAPSHOT/org.openhab.binding.saicismart-3.4.1-SNAPSHOT.jar

### CODE REVIEW

**Comment by tisoft** [comment]
This is not complete, yet. But I would like some in between review, to see if I am going in the right direction.

**Comment by hmerk** [comment]
Please rebase to clean up your PR and remove conflicts and not needed reviewers.

**Comment by hmerk** [comment]
Also, you want to contribute your new binding to the 3.4.x branch. This one is for bugfixes only, development cycle has changed to openHAB 4.0 [main branch] with JDK level 17.

**Comment by tisoft** [comment]
@hmerk Sorry for the reviewer mess...

Can I remove the reviewers somehow?

I want to stay on 3.4 for the time being, since I want to test it with my home automation system, and I don't want that updated to java 17 yet. Also my test users are all still on 3.x right now.

Will rebase against main, if this is feature complete and ready for inclusion.

Marked the PR as draft for the time being.

**Comment by hmerk** [comment]
Main branch is automatically built with JDK 17 and the pom version now says
„openHAB-4.0-SNAPSHOT“.

If you want to stay at 3.4 for testing, built the binding locally and publish it on the marketplace.

Better close this PR and push a new one, once you are ready for openHAB 4.0 and JDK17

File: bundles/org.openhab.binding.saicismart/pom.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.openhab.addons.bundles</groupId>
+    <artifactId>org.openhab.addons.reactor.bundles</artifactId>
+    <version>3.4.1-SNAPSHOT</version>
```
Comment (wborn) [inline bundles/org.openhab.binding.saicismart/pom.xml:10]:
```suggestion
    <version>4.0.0-SNAPSHOT</version>
```

**Comment by jlaur** [comment]
Please create a new PR targeting the **main** branch.

**Pull request closed without merging**